### DISCUSSION TITLE
Why not support threading in pyodide ?

### DISCUSSION THREAD

**Discussion by patrickyyang** [pyodide/pyodide#2223] [Ideas]
I found that pthread is supported by emscripten, so why threading is not supported by pyodide? Are there any traps if I build pyodide by    USE_THREAD ?

**Comment by hoodmane** [comment]
I haven't tried it so I am not sure what will happen. Please try it and let us know what you find out. I think Pyodide will need patching to get the Javascript API working.

Using PTHREADS we are also likely to discover emscripten bugs. We use dynamic linking very heavily and often hit bugs with emscripten's implementation of dynamic linking. As I understand it, dynamic linking and pthreads currently interact poorly -- the function pointer table and other dynamic linking state need to be synchronized across threads but it is difficult to do this in a way that is both correct and reasonably performant.

Of course, it would be highly desirable to discover these bugs and fix them upstream. But I think this may take a large amount of effort. Since Pyodide has a very limited amount of developer effort available, I think adding support for this is currently out of scope.

**Comment by ryanking13** [comment]
Related issue: #237
